### DISCUSSION TITLE
Lexical keeps empty paragraph when everything is removed from the editor

### DISCUSSION THREAD

**Discussion by revnelson** [payloadcms/payload#10103] [Feature Requests & Ideas]
When you remove all content from a lexical editor that previously had content, an empty paragraph is returned from the API.

This creates problems with our frontend serializer adding unwanted whitespace and causing us to have to do ugly checks like:
`if (root?.children?.length === 1 && root.children.at(0).type === 'paragraph' && root.children.at(0).children?.length === 0) return`

Can we get some sort of option to fully remove the `root` object when all content is removed from a lexical editor?

**Comment by wkentdag** [comment]
I made a util to remove the empty node and stuck it in a `beforeChange` hook for any rich text field. It would be nice to have this feature in core but the patch has been working just fine for me.

```ts
// https://stackoverflow.com/questions/75493899/how-to-check-if-the-editor-is-empty
export const isEmpty = (node?: { root: SerializedRootNode }) => {
  if (!node || !node?.root) {
    return true
  }

  const { root } = node
  const isRoot = root.type === 'root'
  const hasSingleChild = root.children.length < 2
  const hasNestedChild = root.children.some((childNode) => {
    // @ts-expect-error @TODO why is `children` not in the typedef?
    if (childNode?.children) {
      // @ts-expect-error saa
      return childNode.children.length > 0
    }

    return false
  })

  return isRoot && hasSingleChild && !hasNestedChild
}
```


👉 [full example](https://gist.github.com/wkentdag/2dccedf9c61cd762789e24f25e9e2f3a)
